crates/primitives-traits/src/extended.rs

```diff
@@ -33,44 +30,41 @@ macro_rules! delegate {
 ///
 /// Note: The other transaction type variants must not overlap with the builtin one, transaction
 /// types must be unique.
-#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Hash, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
 pub enum ExtendedTxEnvelope<BuiltIn, Other> {
+    /// The builtin transaction type.
     BuiltIn(BuiltIn),
+    /// The other transaction type.
     Other(Other),
 }
 
-pub type ExtendedOpTxEnvelope<T> = ExtendedTxEnvelope<OpTxEnvelope, T>;
-
-impl TryFrom<ExtendedTxEnvelope<OpTxEnvelope, CustomTransactionEnvelope>>
-    for ExtendedTxEnvelope<OpPooledTransaction, CustomTransactionEnvelope>
+impl<Tx> TryFrom<ExtendedTxEnvelope<OpTxEnvelope, Tx>>
+    for ExtendedTxEnvelope<OpPooledTransaction, Tx>
```

Comment:
we either need to relax this so that this works for any Builtin type or feature gat behind op feature